doc/connectivity/networking/net_config_guide.rst

```diff
@@ -0,0 +1,267 @@
+  the size of the fixed buffer by setting :kconfig:option:`CONFIG_NET_BUF_DATA_SIZE`, and
+  the size of the data pool size by setting :kconfig:option:`CONFIG_NET_BUF_DATA_POOL_SIZE`
+  if variable size buffers are used.
+
+  When using the fixed size data buffers, the memory consumption of network buffers
+  can be tweaked by selecting the size of the data part according to what kind of network
+  data we are receiving. If one sets the data size to 256, but only receives packets
+  that are 32 bytes long, then we are "wasting" 224 bytes for each packet because we
+  cannot utilize the remaining data. One should not set the data size too low because
+  there is some overhead involved for each net_buf. For these reasons the default
+  network buffer size is set to 128 bytes.
+
+  The variable size data buffer feature is marked as experimental as it has not
+  received as much testing as the fixed size buffers. Using variable size data
+  buffers tries to improve memory utilization by allocating minimum amount of
+  data we need for the network data. The extra cost here is the amount of time
+  that is needed when dynamically allocating the buffer from the memory pool.
+
+  For example, in Ethernet the maximum transmission unit (MTU) size is 1500 bytes.
+  If one wants to receive two full frames, then the net_pkt RX count should be set to 2,
+  and net_buf RX count to (1500 / 128) * 2 which is 24.
+  If TCP is being used, then these values need to be higher because we can queue the
+  packets internally before delivering to the application.
+
+:kconfig:option:`CONFIG_NET_BUF_TX_COUNT`
+  How many network buffers are allocated for sending data. This is similar setting
+  as the receive buffer count but for sending.
+
+
+Connection Options
+******************
+
+:kconfig:option:`CONFIG_NET_MAX_CONN`
+  This option tells how many network connection endpoints are supported.
+  For example each TCP connection requires one connection endpoint. Similarly
+  each listening UDP connection requires one connection endpoint.
+  Also various system services like DHCP and DNS need connection endpoints to work.
+  The network shell command **net conn** can be used at runtime to see the
+  network connection information.
+
+:kconfig:option:`CONFIG_NET_MAX_CONTEXTS`
+  Number of network contexts to allocate. Each network context describes a network
+  5-tuple that is used when listening or sending network traffic. Each BSD socket in the
+  system uses one network context.
+
+
+Socket Options
+**************
+
+:kconfig:option:`CONFIG_NET_SOCKETS_POLL_MAX`
+  Max number of supported poll() entries. One needs to select proper value here depending
+  on how many BSD sockets are polled in the system.
+
+:kconfig:option:`CONFIG_NET_SOCKETPAIR_BUFFER_SIZE`
+  This option is used by socketpair() function. It sets the size of the
+  internal intermediate buffer, in bytes. This sets the limit how large
+  messages can be passed between two socketpair endpoints.
+
```

Comment:
Maybe we should mention `CONFIG_POSIX_MAX_FDS` here as well, as this parameter also limits the number of sockets that can be created in the system.